### PR TITLE
feat: Permit wires bound to methods without param and without return

### LINKED ISSUE #12273: Augment `ComponentWiring` to support no param, no return methods
- In order to facilitate clean migration to proxy wiring, we should have the ability to bind methods that don't accept any parameters and don't return anything
- A good example of this pattern are `clear` methods, that simply need to be triggered by an input wire
- This change will allow us to stop following the ugly "ignored input" pattern

### PR DESCRIPTION
**Pull request by alittley**
- closes #12273

### CODE REVIEW

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java [hunk 1]
```diff
@@ -189,7 +190,28 @@ public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(
...
+     * @return the input wire
+     */
+    public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(@NonNull final Consumer<COMPONENT_TYPE> handler) {
+        Objects.requireNonNull(handler);
+
+        try {
+            handler.accept(proxyComponent);
+        } catch (final NullPointerException e) {
+            throw new IllegalStateException(
+                    "Component wiring does not support primitive input types. Use a boxed primitive instead.");
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java:211]: Minor: might want to put the `e` into the `cause` of your new exception to provide more information when debugging issues. You can never really be sure (now, or in the future as the code grows) what exactly caused an NPE w/o seeing the original exception.

**Review by anthony-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java (hunk 1 above)
Comment (alittley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java]: Added

**Review by anthony-swirldslabs — APPROVED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java [hunk 2]
```diff
@@ -186,10 +188,31 @@ public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(
...
+
+    /**
+     * Get an input wire for this component.
+     *
+     * @param handler      the component method that will handle the input, e.g. "MyComponent::handleInput". Should be a
+     *                     method on the class, not a method on a specific instance.
+     * @param <INPUT_TYPE> the input type
+     * @return the input wire
+     */
+    public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(@NonNull final Consumer<COMPONENT_TYPE> handler) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java:205]:
```suggestion
    @NonNull
    public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(@NonNull final Consumer<COMPONENT_TYPE> handler) {
```

Comment (alittley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java:206]: fixed

**Review by lpetrovic05 — APPROVED** [review]

**Review by anthony-swirldslabs — APPROVED** [review]

**Pull request merged**